packages/jellyfish-api-core/__tests__/category/net/getPeerInfo.test.ts

```diff
@@ -0,0 +1,92 @@
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { PeerInfo } from 'packages/jellyfish-api-core/src/category/net'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+
+createNetworkTests('Network without masternode', new RegTestContainer())
+createNetworkTests('Network on masternode', new MasterNodeRegTestContainer())
+
+function createNetworkTests (name: string, container: RegTestContainer | MasterNodeRegTestContainer): void {
+  describe(name, () => {
+    const client = new ContainerAdapterClient(container)
+
+    beforeAll(async () => {
+      await container.start()
+      await container.addNode(await container.getIp('bridge'))
+    })
+
+    afterAll(async () => {
+      await container.stop()
+    })
+
+    it('should getPeerInfo', async () => {
+      const peers: PeerInfo[] = await client.net.getPeerInfo()
+
+      expect(peers.length).toBeGreaterThan(0)
+
+      for (const peer of peers) {
+        expect(typeof peer.id).toStrictEqual('number')
+
+        expectValidAddress(peer.addr)
+
+        expectUndefinedOrValidAddress(peer.addrbind)
+        expectUndefinedOrValidAddress(peer.addrlocal)
+
+        expect(typeof peer.services).toStrictEqual('string')
+        expect(peer.services).toHaveLength(16)
+
+        expect(typeof peer.relaytxes).toStrictEqual('boolean')
+        expect(typeof peer.lastsend).toStrictEqual('number')
+        expect(typeof peer.lastrecv).toStrictEqual('number')
+        expect(typeof peer.bytessent).toStrictEqual('number')
+        expect(typeof peer.bytesrecv).toStrictEqual('number')
+        expect(typeof peer.conntime).toStrictEqual('number')
+        expect(typeof peer.timeoffset).toStrictEqual('number')
+
+        expectUndefinedOrEquals('number', peer.pingtime)
+        expectUndefinedOrEquals('number', peer.minping)
+        expectUndefinedOrEquals('number', peer.pingwait)
+
+        expect(typeof peer.version).toStrictEqual('number')
+        expect(typeof peer.subver).toStrictEqual('string')
+        expect(typeof peer.inbound).toStrictEqual('boolean')
+        expect(typeof peer.addnode).toStrictEqual('boolean')
+        expect(typeof peer.startingheight).toStrictEqual('number')
+
+        expectUndefinedOrEquals('number', peer.banscore)
+        expectUndefinedOrEquals('number', peer.synced_headers)
+        expectUndefinedOrEquals('number', peer.synced_blocks)
+
+        expect(typeof peer.inflight).toStrictEqual('object')
+        expect(typeof peer.whitelisted).toStrictEqual('boolean')
+        expect(typeof peer.permissions).toStrictEqual('object')
+        expect(typeof peer.minfeefilter).toStrictEqual('number')
+
+        expectValidBytesPerMessage(peer.bytessent_per_msg)
+        expectValidBytesPerMessage(peer.bytesrecv_per_msg)
+      }
+    })
+  })
+}
+
+function expectUndefinedOrEquals (expected: any, actual?: any): void {
+  if (actual !== undefined) {
+    expect(typeof actual).toStrictEqual(expected)
+  }
```

Comment:
the assertion might not be triggered due to the "if"
so if writing this way, we need to use `expect.assections(num)` as well to ensure the "if" is triggered